iosapp/lib-ios/Sources/LibClients/DeviceClient.swift

```diff
@@ -53,32 +56,58 @@ public extension DeviceClient {
     case finished
     case errorCouldNotCreateDir
   }
+
+  struct Data: Sendable {
+    public var type: DeviceType
+    public var iOSVersion: String
+    public var vendorId: UUID?
+
+    public init(type: DeviceType, iOSVersion: String, vendorId: UUID?) {
+      self.type = type
+      self.iOSVersion = iOSVersion
+      self.vendorId = vendorId
+    }
+  }
 }
 
 extension DeviceClient: DependencyKey {
   #if os(iOS)
-    public static let liveValue = DeviceClient(
-      type: UIDevice.current.userInterfaceIdiom == .pad ? .iPad : .iPhone,
-      iOSVersion: UIDevice.current.systemVersion,
-      vendorId: UIDevice.current.identifierForVendor,
-      batteryLevel: {
-        UIDevice.current.isBatteryMonitoringEnabled = true
-        let level = await UIDevice.current.batteryLevel
-        UIDevice.current.isBatteryMonitoringEnabled = false
-        return switch level {
-        case 0.0 ... 1.0: .level(level)
-        default: .unknown
-        }
-      },
-      clearCache: doClearCache,
-      deleteCacheFillDir: { try? FileManager.default.removeItem(at: .fillDir) },
-      availableDiskSpaceInBytes: getAvailableDiskSpaceInBytes
-    )
+    public static var liveValue: DeviceClient {
+      DeviceClient(
+        type: { await MainActor.run {
+          UIDevice.current.userInterfaceIdiom == .pad ? .iPad : .iPhone
+        }},
+        iOSVersion: { await MainActor.run {
+          UIDevice.current.systemVersion
+        }},
+        vendorId: { await MainActor.run {
+          UIDevice.current.identifierForVendor
+        }},
+        data: { await MainActor.run { .init(
```

Comment:
this `data()` field is just a convenience for getting all (except battery) main actor info in one swoop, to avoid 3 thread hops to the main thread, since these are often needed together (for api logging, etc.)